cmd/kubeadm/app/cmd/phases/init/certs.go

```diff
@@ -197,6 +199,20 @@ func runCerts(c workflow.RunData) error {
 	}
 
 	fmt.Printf("[certs] Using certificateDir folder %q\n", data.CertificateWriteDir())
+
+	// If using an external CA while dryrun, copy CA cert to dryrun dir for later use
+	if data.ExternalCA() && data.DryRun() {
+		externalCAFile := filepath.Join(data.Cfg().CertificatesDir, kubeadmconstants.CACertName)
+		fileInfo, _ := os.Stat(externalCAFile)
+		contents, err := os.ReadFile(externalCAFile)
+		if err != nil {
+			return err
+		}
+		err = os.WriteFile(filepath.Join(data.KubeConfigDir(), kubeadmconstants.CACertName), contents, fileInfo.Mode())
```

Comment:
should we use CertificateDir() here?
technically the CA is not a kubeconfig file.

perhaps modify:
https://github.com/kubernetes/kubernetes/blob/e9a0ca8ff7c1c6bd021a1f8a6ebfec618dd8f7c9/cmd/kubeadm/app/cmd/init.go#L474-L476

to return:
```
if d.dryRun {
		return d.dryRunDir
	}
```

Reply:
YES, actually there is a CertificateWriteDir() design for this situation and I should use that.
![image](https://user-images.githubusercontent.com/39937150/124264880-9dc54e00-db67-11eb-8b9a-ed0bb1591f0d.png)
Fixed.